irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcCallbackController.java

```diff
@@ -48,19 +57,50 @@ public class EdcCallbackController {
     private final EndpointDataReferenceStorage storage;
 
     @PostMapping
-    public void receiveEdcCallback(final @RequestBody EndpointDataReference dataReference) {
-        log.debug("Received EndpointDataReference: {}", StringMapper.mapToString(dataReference));
-        log.debug("Received EndpointDataReference with ID {} and endpoint {}", dataReference.getId(),
-                dataReference.getEndpoint());
-        final String authCode = dataReference.getAuthCode();
-        if (authCode != null) {
-            final var contractAgreementId = EDRAuthCode.fromAuthCodeToken(authCode).getCid();
-            storage.put(contractAgreementId, dataReference);
-            log.info("Endpoint Data Reference received and cached for agreement: {}", Masker.mask(contractAgreementId));
-        } else {
-            log.error("ContractAgreementId could not be extracted from Endpoint Data Reference {}",
-                    StringMapper.mapToString(dataReference));
+    public void receiveEdcCallback(final @RequestBody String endpointDataReferenceCallback) {
+        EndpointDataReference endpointDataReference;
+
+        try {
+            endpointDataReference = mapToEndpointDataReference(endpointDataReferenceCallback);
+
+            log.debug("Received EndpointDataReference: {}", StringMapper.mapToString(endpointDataReference));
+            log.debug("Received EndpointDataReference with ID {} and endpoint {}", endpointDataReference.getId(),
+                    endpointDataReference.getEndpoint());
+
+            final String contractId = endpointDataReference.getContractId();
+            storeEdr(contractId, endpointDataReference);
+        } catch (EdcClientException e) {
+            log.error("Could not deserialize Endpoint Data Reference {}", endpointDataReferenceCallback);
         }
+
+    }
+
+    private static EndpointDataReference mapToEndpointDataReference(final String endpointDataReference)
+            throws EdcClientException {
+        try {
+            EndpointDataReference dataReference;
```

Comment:
```suggestion
            final EndpointDataReference dataReference;
```
Furthermore declaration should be outside the try just like it is done in the other method.
Reason: Makes it clear at once that this variable contains the methods results.

Reply:
done